client/my-sites/site-settings/form-general.jsx

```diff
@@ -286,48 +381,50 @@ const FormGeneral = React.createClass( {
 						<FormRadio
 							name="blog_public"
 							value="-1"
-							checked={ -1 === parseInt( this.state.blog_public, 10 ) }
+							checked={ -1 === parseInt( fields.blog_public, 10 ) }
 							onChange={ this.handleRadio }
-							disabled={ this.state.fetchingSettings }
+							disabled={ isRequestingSettings }
 							onClick={ this.onRecordEvent( 'Clicked Site Visibility Radio Button' ) } />
-						<span>{ this.translate( 'Private' ) }</span>
+						<span>{ translate( 'Private' ) }</span>
 						<FormSettingExplanation isIndented>
-							{ this.translate( 'Your site is only visible to you and users you approve.' ) }
+							{ translate( 'Your site is only visible to you and users you approve.' ) }
 						</FormSettingExplanation>
 					</FormLabel>
 				}
 
 			</FormFieldset>
 		);
-	},
+	}
 
-	handleAmpToggle() {
-		this.setState( { amp_is_enabled: ! this.state.amp_is_enabled }, () => {
+	handleAmpToggle = () => {
+		const { fields, updateFields } = this.props;
+		updateFields( { amp_is_enabled: ! fields.amp_is_enabled }, () => {
 			this.submitForm();
 			this.onRecordEvent( 'Clicked AMP Toggle' );
 		} );
-	},
+	};
 
-	handleAmpCustomize() {
+	handleAmpCustomize = () => {
 		this.onRecordEvent( 'Clicked AMP Customize button' );
 		page( '/customize/amp/' + this.props.site.slug );
-	},
+	};
 
 	renderAmpSection() {
-		const { site } = this.props;
-
-		if ( site.jetpack ) {
+		if ( this.props.site.jetpack ) {
 			return;
 		}
 
 		const {
-			fetchingSettings,
-			submittingForm,
-			amp_is_supported,
-			amp_is_enabled,
-		} = this.state;
-
-		const isDisabled = fetchingSettings || submittingForm;
+			fields: {
+				amp_is_supported,
```

Comment:
Minor: While it's admittedly a pain, I don't think we should be declaring our own variables that don't adhere to `camelCase` guidelines, even via destructuring. Instead this could be `amp_is_supported: ampIsSupported` if we wanted to use it directly.